lib/Optimized64/KeccakP-1600-AVX512-plainC.c

```diff
@@ -213,29 +213,39 @@ void KeccakP1600_AVX512_Permute_12rounds(void *state)
 /* ---------------------------------------------------------------- */
 
 #include <assert.h>
+#include <stdio.h>
 
 size_t KeccakP1600_AVX512_12rounds_FastLoop_Absorb(void *state, unsigned int laneCount, const unsigned char *data, size_t dataByteLen)
 {
     size_t originalDataByteLen = dataByteLen;
 
-    assert(laneCount == 21);
+    assert(laneCount == 21 || laneCount == 17);
 
     KeccakP_DeclareVars;
     uint64_t *stateAsLanes = (uint64_t*)state;
     uint64_t *inDataAsLanes = (uint64_t*)data;
 
-    copyFromState(stateAsLanes);
-    while(dataByteLen >= 21*8) {
-        Baeiou = XOR(Baeiou, LOAD_Plane(inDataAsLanes+ 0));
-        Gaeiou = XOR(Gaeiou, LOAD_Plane(inDataAsLanes+ 5));
-        Kaeiou = XOR(Kaeiou, LOAD_Plane(inDataAsLanes+10));
-        Maeiou = XOR(Maeiou, LOAD_Plane(inDataAsLanes+15));
-        Saeiou = XOR(Saeiou, LOAD_Lane(inDataAsLanes+20));
-        rounds12;
-        inDataAsLanes += 21;
-        dataByteLen -= 21*8;
+    if (laneCount == 21) {
+        copyFromState(stateAsLanes);
+        while(dataByteLen >= 21*8) {
+            Baeiou = XOR(Baeiou, LOAD_Plane(inDataAsLanes+ 0));
+            Gaeiou = XOR(Gaeiou, LOAD_Plane(inDataAsLanes+ 5));
+            Kaeiou = XOR(Kaeiou, LOAD_Plane(inDataAsLanes+10));
+            Maeiou = XOR(Maeiou, LOAD_Plane(inDataAsLanes+15));
+            Saeiou = XOR(Saeiou, LOAD_Lane(inDataAsLanes+20));
+            rounds12;
+            inDataAsLanes += 21;
+            dataByteLen -= 21*8;
+        }
+        copyToState(stateAsLanes);
+    } else if (laneCount == 17) {
+        while(dataByteLen >= laneCount*8) {
```

Comment:
How difficult would it be to transpose the method for 21 lanes to 17 lanes here?

Reply:
I am honestly not sure how the method for 21 lanes works, I can go into it in more details, and then try to adapt it for 17 lanes. It would be appreciated if you could recommend any resources to refer to that would help in understanding how it works. Thank you :)